kernelboard/static/news/2025-09-10-jane-street-hackathon.md

```diff
@@ -0,0 +1,53 @@
+</figure>
+
+Participants came from all across North America, (California, Wisconsin, Texas, Toronto, etc.), and were both students and industry vets such as CUDA programmers from trading firms or folks working on GPU profilers. Overall, there was overwhelming positive feedback from [attendees](https://x.com/bdepyzy_/status/1965121181503422943?s=46), [winners](https://x.com/NadavTimor/status/1965082710730551701) and even [those searching for talent](https://x.com/stake_jevens/status/1964499440133427256) with a day that had participants [so locked in](https://x.com/GPU_MODE/status/1964516306822045996) that many opted to skip Jane Street’s delicious spread in favour of optimizing their models.
+
+![Final Moments](/static/images/nailbiting_JS.png)
+
+A big congratulations to the winning teams of each round - Cougar & [Crane](https://www.linkedin.com/posts/kylecyu_we-won-1st-place-at-the-jane-street-x-gpu-activity-7370837082140774400-dvxP?utm_source=share&utm_medium=member_desktop&rcm=ACoAAAWbErgBtCPmRJStcIqOGtOiTogXhv9YJ8g)!
+
+![Team Cougar](/static/images/cougar_JS.png)
+![Team Crane](/static/images/crane_JS.png)
+
+###### The Submissions
+[This submission](https://github.com/kyolebu/janestreet-gpumode-hackathon/blob/main/example_model.py) produced the highest PnL in round 2 and made the organizers most proud. The main “kernel” level optimization was actually just running `self.model = torch.compile(self.model,mode="reduce-overhead",fullgraph=True)`. The biggest lift came from batching incoming symbols such that you could process multiple symbols at once.  Doing batching correctly was extremely difficult as one needed to keep track of the models’ hidden state per symbol. Only 2-3 teams managed to do this correctly, and the winning team only managed to do it in the last hour of competition or so (and the game authors could barely do it in time during play testing).
+
+Team strategies tended to range. Some folks ignored the hidden state or portions of the model entirely. While this strategy reduced the model’s accuracy, it greatly reduced latency. Others played around with torch.compile flags and did various quantization schemes (mostly to fp16/bf16). Two teams used distillation in their strategies. The team who committed fully to it managed to get a model ~300% faster than the winning team, but at the cost of being right only a fifth of the time. Another team tried to make use of all 6 GPUs they were given (usually a team used 1 to serve their model and 5 for development), and attempted to do distributed inference.  Overall, we were happy with the mix of strategies, and thought there was a healthy amount of experimentation. Notably, writing custom kernels did not contribute to a team winning. Also worth noting is that to those new to them, it was very nonobvious what the torch.compile flags were or what would cause a graph break/segfaults.
```

Comment:
>  Notably, writing custom kernels did not contribute to a team winning. Also worth noting is that to those new to them, it was very nonobvious what the torch.compile flags were or what would cause a graph break/segfaults.

This was more so feedback for pytorch folks. The second sentence isn't very relevant to those outside of pytorch, please remove it. 

You can remove the first as well or say something more akin to no one successfully wrote a kernel due to time pressure. (but I'm ambivalent).